cmd/golangci-lint/main.go

```diff
@@ -3,20 +3,43 @@ package main
 import (
 	"fmt"
 	"os"
+	"runtime/debug"
 
 	"github.com/golangci/golangci-lint/pkg/commands"
 	"github.com/golangci/golangci-lint/pkg/exitcodes"
 )
 
 var (
+	goVersion = "unknown"
+
 	// Populated by goreleaser during build
 	version = "master"
 	commit  = "?"
 	date    = ""
 )
 
+//nolint:gochecknoinits
+func init() {
+	if info, available := debug.ReadBuildInfo(); available {
+		goVersion = info.GoVersion
+
+		if date == "" {
+			version = info.Main.Version
+			commit = fmt.Sprintf("(unknown, mod sum: %q)", info.Main.Sum)
+			date = "(unknown)"
+		}
+	}
+}
+
 func main() {
-	e := commands.NewExecutor(version, commit, date)
+	info := commands.BuildInfo{
```

Comment:
Could we add an integration test that checks the output of running commands:
```
golangci-lint version
golangci-lint run --version
golangci-lint version --format json
golangci-lint version --debug
```
?

Reply:
no, because it will depend on dependencies, the tests will be weak.